File: quick-build/corpus/apps/sora-editor-lib/src/main/java/io/github/rosemoe/sora/corpusharness/EditorHostActivity.java
package io.github.rosemoe.sora.corpusharness;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;
import io.github.rosemoe.sora.text.ContentLine;

/**
 * Harness scaffolding for the quick-build corpus (ADFA-4128 D4) -- NOT part of upstream sora-editor. Exercises real editor-core content (ContentLine, LineSeparator) without pulling in CodeEditor's full transitive dependency graph (which this corpus tier deliberately excludes -- see README.md "sora-editor-lib" for why). Written in Java (not Kotlin) so it can reference the vendored Java classes directly -- the daemon's Kotlin-first compile pass has no visibility into co-located Java sources (see README).
 */
public class EditorHostActivity extends Activity {

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		ContentLine line = new ContentLine(SampleText.INITIAL);
		TextView view = new TextView(this);
		view.setText(line.toString());
		setContentView(view);
	}
}

File: quick-build/corpus/apps/sora-editor-full/edits/02-kotlin-body-edit/EditorHostActivity.kt
package org.appdevforall.corpusharness

import android.app.Activity
import android.os.Bundle
import io.github.rosemoe.sora.widget.CodeEditor

/**
 * Harness scaffolding for the quick-build corpus (ADFA-4128 WS-D) - NOT part of upstream
 * sora-editor.
 *
 * Deliberately KOTLIN referencing the Java [CodeEditor]: that is the exact direction the
 * corpus README's large-real-app finding 2 reported as impossible ("Unresolved reference
 * 'CodeEditor' in Kotlin sources"), so this file is the reproduction as much as it is an
 * entry point.
 */
class EditorHostActivity : Activity() {
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		val editor = CodeEditor(this)
		editor.setText("QB_SORA_KBODY_MARKER_V2")
		setContentView(editor)
	}
}
